react/src/base/inputs/SprkTextInput/SprkTextInput.js

```diff
@@ -168,29 +171,24 @@ SprkTextInput.propTypes = {
    */
   disabled: PropTypes.bool,
   /**
-   * A space-separated string of valid HTML ids to add to the aria-describedby
-   * attribute on the Input.
+   * Assigned to the `aria-describedby`
+   * attribute. Used to create
+   * relationships between the
+   * component and text that describes it,
+   * such as helper text or an error field.
    */
   ariaDescribedBy: PropTypes.string,
+  type: PropTypes.string,
+  value: PropTypes.string,
 };
 
 SprkTextInput.defaultProps = {
-  additionalClasses: '',
-  analyticsString: '',
-  children: [],
-  errorMessage: '',
   formatter: (value) => value,
   forwardedRef: React.createRef(),
   helperText: '',
-  hiddenLabel: false,
-  iconRight: false,
-  idString: '',
   label: 'Text Input Label',
   leadingIcon: '',
```

Comment:
Did this need to stay for a reason?

Reply:
Yes, there isn't a check for when there's no leadingIcon t so it will throw an error. I didn't update the component to add a check because it's going to be deprecated.